helpers/update-site/resources.js

```diff
@@ -0,0 +1,318 @@
+ * Before: 'hint-amp-validator'
+ * After:
+ * {
+ *     isSummary: false,
+ *     link: '/docs/user-guide/hints/hint-amp-validator/'
+ *     name: 'amp-validator'
+ * }
+ */
+const processHint = (hint, isSummary) => {
+    const processedHint = {
+        /*
+         * For packages with multiple hints we have a summary
+         * file with how to install and links to all the hints.
+         * This property is used to not take into account in
+         * the hint index page total.
+         */
+        isSummary: isSummary === true,
+        link: `/docs/user-guide/hints/${hint}/`,
+        name: hint.replace(/^hint-/, '')
+    };
+
+    return processedHint;
+};
+
+/**
+ * Process catogories to be suitable for Hexo.
+ * Before: { performance: ['hint-amp-validator'] }
+ * After:
+ * [{
+ *      name: 'performance',
+ *      hints: [{
+ *          link: 'hint-amp-validator/',
+ *          name: 'amp-validator'
+ *      }]
+ * }]
+ */
+const processCategories = (cats) => {
+    const processedCategories = _.reduce(cats, (allCategories, category) => {
+        const includedHints = category.hints;
+
+        const singleHints = includedHints.filter((hintName) => {
+            return !hintName.includes('/');
+        });
+
+        /*
+         * Hints with id: package-name/hint-name will be
+         * considered part of a package with multiple hints
+         */
+        const multiHints = includedHints.filter((hintName) => {
+            return hintName.includes('/');
+        });
+
+        /*
+         * Group hints by package name.
+         */
+        const packagesName = _.groupBy(multiHints, (hintName) => {
+            return hintName.split('/')[0];
+        });
+
+        const hints = _.map(singleHints, processHint);
+
+        const multiHintsProcessed = _.reduce(packagesName, (multi, values, key) => {
+            // Add an item with the link to the package with multiple hints itself.
+            const partial = processHint(key, true);
+
+            multi.push(partial);
+
+            // Add an item for each individual hint for a package with multiple hints.
+            return multi.concat(_.map(values, processHint));
+        }, []);
+
+        const processedCategory = {
+            description: category.description,
+            hints: hints.concat(multiHintsProcessed),
+            link: `/docs/user-guide/hints/${category.normalizedName}/`,
+            name: category.name,
+            normalizedName: category.normalizedName
+        };
+
+        allCategories.push(processedCategory);
+
+        return allCategories;
+    }, []);
+
+    return { categories: processedCategories };
+};
+
+const generateHintCategoryDocAndInfo = () => {
+    const configDir = path.resolve(__dirname, '..', '..', '_config.yml');
+    const hexoConfig = yamlLoader.safeLoad(fs.readFileSync(configDir, 'utf8')); // eslint-disable-line no-sync
+    /*
+     * We are reading the metadata information directly from the TS files using regex.
```

Comment:
Not blocking, but since we're doing this should we consider getting the user-friendly display name from here as well instead of from `README.md`? My only hesitation is due to multi-hints since there isn't metadata for the group, but I'd be curious to hear your thoughts.

Reply:
Yes, but right know, the user-friendly display is not calculated here, is done in another "task" later in the build process. That means, that we should read a couple of times the file to get later the title.

Once that said, I like the idea of taking the name from the metadata too. If you and @molant are ok, I can dedicate more time to this to try to make it a littler bit better.

Reply:
I'm ok dedicanting more time